interview-questions.md

```diff
@@ -8,62 +8,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is iteration?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer:Iteration is number of execution in a loop until a certain condition met.
 
+  Researched answer:Iteration is reapting a certain steps or instruction until a ceratin condition is met.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer:the predefined parameters taken by map higher order function are 
+  value ,index and array.
 
-  Researched answer:
+  Researched answer:Generally all built in higher order methods take three parameters and the y are value,index and an array.
```

Comment:
Good work, please use the research answer area to dig deeper into a subject. Find something that will impress your interviewer.